FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-LTS-Beta2/c_sdk/platform/freertos/transport/src/mbedtls_freertos.c

```diff
@@ -0,0 +1,541 @@
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Initialize the mbed TLS structures in a network connection.
+ *
+ * @param[in] pSslContext The SSL context to initialize.
+ */
+static void sslContextInit( SSLContext_t * pSslContext );
+
+/**
+ * @brief Free the mbed TLS structures in a network connection.
+ *
+ * @param[in] pSslContext The SSL context to free.
+ */
+static void sslContextFree( SSLContext_t * pSslContext );
+
+/**
+ * @brief Set up TLS on a TCP connection.
+ *
+ * @param[in] pNetworkContext Network context.
+ * @param[in] pHostName Remote host name, used for server name indication.
+ * @param[in] pNetworkCredentials TLS setup parameters.
+ *
+ * @return #MBEDTLS_SUCCESS, #MBEDTLS_INSUFFICIENT_MEMORY, #MBEDTLS_INVALID_CREDENTIALS,
+ * #MBEDTLS_HANDSHAKE_FAILED, or #MBEDTLS_API_ERROR.
+ */
+static MbedtlsStatus_t tlsSetup( NetworkContext_t * pNetworkContext,
+                                 const char * pHostName,
+                                 const NetworkCredentials_t * pNetworkCredentials );
+
+/**
+ * @brief Initialize mbedTLS.
+ *
+ * @return #MBEDTLS_SUCCESS, or #MBEDTLS_API_ERROR.
+ */
+static MbedtlsStatus_t initMbedtls( void );
+
+/*-----------------------------------------------------------*/
+
+static void sslContextInit( SSLContext_t * pSslContext )
+{
+    mbedtls_ssl_config_init( &( pSslContext->config ) );
+    mbedtls_x509_crt_init( &( pSslContext->rootCa ) );
+    mbedtls_pk_init( &( pSslContext->privKey ) );
+    mbedtls_x509_crt_init( &( pSslContext->clientCert ) );
+    mbedtls_ssl_init( &( pSslContext->context ) );
+}
+/*-----------------------------------------------------------*/
+
+static void sslContextFree( SSLContext_t * pSslContext )
+{
+    mbedtls_ssl_free( &( pSslContext->context ) );
+    mbedtls_x509_crt_free( &( pSslContext->rootCa ) );
+    mbedtls_x509_crt_free( &( pSslContext->clientCert ) );
+    mbedtls_pk_free( &( pSslContext->privKey ) );
+    mbedtls_ssl_config_free( &( pSslContext->config ) );
+}
+
+/*-----------------------------------------------------------*/
+
+static MbedtlsStatus_t tlsSetup( NetworkContext_t * pNetworkContext,
+                                 const char * pHostName,
+                                 const NetworkCredentials_t * pNetworkCredentials )
+{
+    MbedtlsStatus_t returnStatus = MBEDTLS_SUCCESS;
+    int mbedtlsError = 0;
+
+    /* Initialize the mbed TLS context structures. */
+    sslContextInit( &( pNetworkContext->sslContext ) );
+
+    mbedtlsError = mbedtls_ssl_config_defaults( &( pNetworkContext->sslContext.config ),
+                                                MBEDTLS_SSL_IS_CLIENT,
+                                                MBEDTLS_SSL_TRANSPORT_STREAM,
+                                                MBEDTLS_SSL_PRESET_DEFAULT );
+
+    if( mbedtlsError != 0 )
+    {
+        LogError( ( "Failed to set default SSL configuration, error %d.", mbedtlsError ) );
```

Comment:
The mbed TLS error codes should be printed as strings using the custom utility functions present [here](https://github.com/FreeRTOS/FreeRTOS/blob/lts-development/FreeRTOS-Plus/Source/mbedtls_utils/mbedtls_error.h#L34-L54) in the repository